### PR TITLE
Add tcalls sched_setscheduler, sched_getscheduler, sched_getparam, sched_get_priority_max, sched_get_priority_min

### PR DESCRIPTION
**Pull request by RRathna**
To-do:
1. enable ltp tests
 - sched_get_priority_min02/max02 pass. All the other tests fail.
2. Explore security implication of sending out sched_setscheduler call to the host.

### CODE REVIEW

**Review by paulcallen — CHANGES_REQUESTED** [review]
There will need tom be some parameter validation on the structure parameters in the kernel, and need to enable the LTP tests for these if possible.

File: kernel/procfs.c [hunk 1]
```diff
@@ -97,6 +101,25 @@ int procfs_pid_setup(pid_t pid)
         ECHECK(myst_mkdirhier(locals->tmp_path, 777));
     }
 
+    /* Create /proc/sys/kernel directory */
```
Comment (vtikoo) [inline kernel/procfs.c:104]: /proc/sys/kernel is not a pid specific entry. `procfs_pid_setup` is called on process creation.

Comment (vtikoo) [reply kernel/procfs.c:104]: Recommend moving this to `create_proc_root_entries`.

File: kernel/procfs.c [hunk 2]
```diff
@@ -97,6 +101,25 @@ int procfs_pid_setup(pid_t pid)
...
+    {
+        ECHECK(myst_snprintf(
+            locals->tmp_path, sizeof(locals->tmp_path), "/proc/sys/kernel"));
+
+        ECHECK(myst_mkdirhier(locals->tmp_path, 777));
+
+        /* Create /proc/sys/kernel/pid_max entry */
+        ECHECK(myst_snprintf(
```
Comment (vtikoo) [inline kernel/procfs.c:112]: You don't need the string formatting, you could directly use the string literal in `myst_create_virtual_file`.

File: kernel/procfs.c [hunk 3]
```diff
@@ -97,6 +101,25 @@ int procfs_pid_setup(pid_t pid)
...
+            locals->tmp_path,
+            sizeof(locals->tmp_path),
+            "/proc/sys/kernel/pid_max"));
```
Comment (vtikoo) [inline kernel/procfs.c:115]: This should be a path within /proc mount. So just "/sys/kernel/pid_max".

**Comment by radhikaj** [comment]
Closing this for now, please reopen if you would like to update

**Pull request closed without merging**